src/sql/workbench/browser/modelComponents/text.component.ts

```diff
@@ -226,6 +228,29 @@ export default class TextComponent extends TitledComponent<azdata.TextComponentP
 		element.style.fontWeight = this.CSSStyles['font-weight']?.toString();
 		return element;
 	}
+
+	public showTooltip(e: Event): void {
+		const descriptionDiv = <HTMLElement>e.target;
+		const tooltip = <HTMLElement>(descriptionDiv.querySelector('.modelview-text-tooltip-content'));
+		tooltip.style.display = '';
+	}
+
+	public onDescriptionKeyDown(e: Event): void {
+		if (e instanceof KeyboardEvent) {
+			let event = new StandardKeyboardEvent(e);
+			const descriptionDiv = <HTMLElement>e.target;
+			const tooltip = <HTMLElement>(descriptionDiv.querySelector('.modelview-text-tooltip-content'));
+			if (event.equals(KeyCode.Escape)) {
+				tooltip.style.display = 'none';
+				event.stopPropagation();
```

Comment:
FYI - it's preferred to use [EventHelper.stop(e, true)](https://github.com/Microsoft/azuredatastudio/blob/d23edbb3380911a072ad344a8f0ea02079e67d27/src/vs/base/browser/dom.ts#L887) for stopping propagation and preventing the default. Easier and handles browser edge cases.